packages/angular_devkit/build_angular/src/angular-cli-files/utilities/check-port.ts

```diff
@@ -5,26 +5,36 @@
  * Use of this source code is governed by an MIT-style license that can be
  * found in the LICENSE file at https://angular.io/license
  */
+import * as net from 'net';
 
-import { Observable } from 'rxjs';
-const portfinder = require('portfinder');
+export function checkPort(port: number, host: string, basePort = 49152): Promise<number> {
+  return new Promise<number>((resolve, reject) => {
+    function _getPort(portNumber: number) {
+      if (portNumber > 65535) {
+        reject(new Error(`There is no port available.`));
+      }
 
+      const server = net.createServer();
 
-export function checkPort(port: number, host: string, basePort = 49152): Observable<number> {
-  return new Observable(obs => {
-    portfinder.basePort = basePort;
-    // tslint:disable:no-any
-    portfinder.getPort({ port, host }, (err: any, foundPort: number) => {
-      if (err) {
-        obs.error(err);
-      } else if (port !== foundPort && port !== 0) {
-        // If the port isn't available and we weren't looking for any port, throw error.
-        obs.error(`Port ${port} is already in use. Use '--port' to specify a different port.`);
-      } else {
-        // Otherwise, our found port is good.
-        obs.next(foundPort);
-        obs.complete();
-      }
-    });
+      server.once('error', err => {
+        if (err.code !== 'EADDRINUSE') {
+          reject(err);
+        } else if (port === 0) {
+          _getPort(portNumber + 1);
+        } else {
+          // If the port isn't available and we weren't looking for any port, throw error.
+          reject(new Error(`
```

Comment:
this will print addition break lines in the Error and potentially in the console. Use `tags.stripIndent`

Ex
```
reject(
    new Error(`Port ${port} is already in use. Use '--port' to specify a different port.`)
);
```

or

tags.stripIndent
```
reject(new Error(tags.stripIndent`
     Port ${port} is already in use. Use '--port' to specify a different port.
`));
```

Reply:
This fails on lint check.

Reply:
Add comma add the end.

Reply:
Done!